src/OrchardCore/OrchardCore.Environment.Shell.Abstractions/Builders/ShellContext.cs

```diff
@@ -120,14 +148,18 @@ protected virtual void Dispose(bool disposing)
 
         internal class ServiceScopeWrapper : IServiceScope
         {
+            private readonly ShellContext _shellContext;
             private readonly IServiceScope _serviceScope;
             private readonly IServiceProvider _existingServices;
             private readonly HttpContext _httpContext;
 
-            public ServiceScopeWrapper(IServiceScope serviceScope)
+            public ServiceScopeWrapper(ShellContext shellContext)
             {
-                ServiceProvider = serviceScope.ServiceProvider;
-                _serviceScope = serviceScope;
+                shellContext.ScopeStarted();
```

Comment:
Nice change. This should address https://github.com/OrchardCMS/OrchardCore/pull/1602#discussion_r173884565.

Reply:
Thanks, i hope, i will take a look on your related comment.